### ISSUE TITLE
Pylance incorrectly displays non-printable bytes literals as `b'\uxxxx'`

### ISSUE THREAD

**Issue by willfrey** [microsoft/pyright#4479]
Pylance is incorrectly converting and displaying a byte literal that contains a non-printable ASCII character as its unicode escape sequence. This is fine for string literals, where Pylance will show

```python3
RS = "\x1e"  # Pylance will infer this has type `Literal["\u001e"]
```

but this is not correct for bytes literals

```python3
RS = b"\x1e"  # Pylance will infer this has type `Literal[b'\u001e']
```

[`\uxxxx` is only supported in string literals according to the docs.](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals)

This seems to be the case for all non-printable ASCII character that do not have a shorthand form in string literals or bytes literals in Python, such as `"\n"` and `"\t"`.

<h3> Environment data </h3>

-   Language Server version: 2023.1.20
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.10.9
-   python.analysis.indexing: null
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

```python3
RS = b"\x1e"
reveal_type(RS)  # Type of "RS" is Literal[b'\u001e']
```

<h3> Expected behavior </h3>

This is what Jedi displays:

![Screenshot 2023-01-17 at 12 47 40 PM](https://user-images.githubusercontent.com/13784361/212975775-03e9c226-a66b-4596-8b2a-d3c8322fd40d.png)

<h3> Actual behavior </h3>

And this is what Pylance displays, as of this writing:

![Screenshot 2023-01-17 at 12 46 47 PM](https://user-images.githubusercontent.com/13784361/212975820-c4135971-130e-4f22-b880-35582ddf34fe.png)

Thank you!

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.290, which I just published. It will also be included in a future release of pylance.
